sources/updater/update_metadata/src/lib.rs

```diff
@@ -320,34 +319,40 @@ impl Update {
         }
     }
 
-    pub fn update_ready(&self, seed: u32) -> bool {
-        // Has this client's wave started
-        if let Some(wave) = self.update_wave(seed) {
-            return wave.has_started();
-        }
-
-        // Or there are no waves
-        true
-    }
-
-    pub fn jitter(&self, seed: u32) -> Option<DateTime<Utc>> {
+    /// Returns whether the update is truly available. A update is said to be 'ready/available' if the wave
+    /// this host belongs to has fully passed, or if the host's position in the wave has passed.
+    /// The position of the host within the wave is determined by the seed value.
+    pub fn update_ready(&self, seed: u32, time: DateTime<Utc>) -> bool {
+        // If this host is part of some update wave
         if let Some(wave) = self.update_wave(seed) {
             if wave.has_passed() {
-                return None;
+                return true;
             }
-            let bounds = match self.update_wave(seed) {
-                Some(Wave::Initial { end }) => Some((Utc::now(), end)),
-                Some(Wave::General { start, end }) => Some((start, end)),
-                Some(Wave::Last { start: _ }) | None => None,
+            let bound = match wave {
+                // For all intent and purposes, with the way we define update wave structures,
+                // the initial wave (0th wave) never has any updates.
+                // No seed value would ever land the host in this initial wave. But if for whatever reason
+                // this host is part of the initial (0th) wave, the update is going to be available
+                Wave::Initial { .. } => None,
+                Wave::General { start, end } => Some((start, Some(end))),
+                Wave::Last { start } => Some((start, None)),
             };
-            if let Some((start, end)) = bounds {
-                let mut rng = thread_rng();
-                if let Some(range) = end.timestamp().checked_sub(start.timestamp()) {
-                    return Some(start + Duration::seconds(rng.gen_range(1, range)));
+            if let Some((start, end)) = bound {
+                if let Some(end) = end {
+                    let wave_duration = end - start;
+                    let time_increment = wave_duration / MAX_SEED as i32;
+                    // Derive the target time position within the wave given the host's seed.
+                    let target_time = start + (time_increment * (seed as i32));
+                    // If the current time is past the target time position in the wave, the update is
+                    // marked available
+                    return time >= target_time;
+                } else {
+                    // Check if last wave has started
+                    return time >= start;
```

Comment:
Yeah, even before this change, the hosts part of the last wave will always all update at once.

That would mean we would always need to define the duration of the last wave in our `waves.toml` which currently we do not. 

I think I'll go ahead and define the duration of the last wave to be 1 day.

So for example, if the last wave has 2048-512=1536 seeds allocated to it (going off of our [`default-waves.toml`](https://github.com/bottlerocket-os/bottlerocket/blob/develop/sources/updater/waves/default-waves.toml)), then the time increment for each seed would be (86400 seconds / 1536) ~= 1 minute. Does that sound reasonable? @tjkirch

Reply:
I don't love the idea of using an arbitrary time period for the last wave - I'd rather clean up our understanding of wave definitions, which is currently fairly murky.  I've already seen people interpreting the wave times differently.  The "initial" wave is weird because it shouldn't actually get used, the final wave is weird because of this jump, intermediate waves still have questions about whether the time is start vs. end, etc.

Until we fix that, I think we could set an explicit ramp for the last wave in our wave TOML files by adding another wave at the end for 99/100%.